### ISSUE TITLE
Provide a vmservice method for getting the package config url for an isolate(group)

### ISSUE THREAD

**Issue by sigurdm** [dart-lang/sdk#56784]
There seems to be no such method yet

Currently devtools is jumping through a lot of hoops to do this. It looks for package configs from the library containing the main file of the current root isolate. 

This is not always correct (the main file might be exported from an imported package)

Also finding the .dart_tool/package_config.json via the dtd protocol is also troublesome, given that it has limited file access (cannot reach files outside the IDE root, which is not guaranteed to contain the package config if the ide is scoped to only parts of a pub workspace.)

Here is the current code:

https://github.com/flutter/devtools/blob/b77c98038700521c4b6da9183fd356674bd42a1a/packages/devtools_shared/lib/src/utils/file_utils.dart#L31

**Comment by mkustermann** [comment]
The concept of a package root may not actually exist. For example if an App was AOT compiled and run on the VM in AOT mode there's no source code anymore, no package dependencies, just an AOT-compiled ELF file the VM can run (profile builds are AOT and can also be inspected via vm-service)

Then there's the question if the VM runs on another device (e.g. in android phone) and the code is pushed to it in the form of a kernel file. Then there may also not be a concept of package root anymore (that's a concept on the host device that compiled to kernel).

So the API would only work for a subset of scenarios. We already expose this programmatically via `dart:isolate`s `Isolate.packageConfigSync` property (which works in those subset of scenarios).

Could one just invoke this getter via `vm-service` protocol?

**Comment by sigurdm** [comment]
> The concept of a package root may not actually exist.

Good point. I wonder if we need service extensions when debugging aot compiled binaries... I guess we do.

> Could one just invoke this getter via vm-service protocol?

Possibly - can we do this while the debugged code is running?

**Comment by sigurdm** [comment]
Maybe the dtd protocol should be the one providing this...

**Comment by sigurdm** [comment]
cc @kenzieschmoll

**Comment by a-siva** [comment]
//cc @bkonyi (something to look into for VM service)
